packages/common/src/utils/objects.ts

```diff
@@ -17,20 +17,14 @@ export const applyDefaults = <TObject>(obj: TObject, defaults: TObject): TObject
 export const applyOverrides = <TObject>(obj: TObject, overrides: TObject): TObject =>
   _.defaultsDeep({}, overrides, obj);
 
-/**
- * Return `true` if `obj` is a plain non-React object.
- */
-export const isPlainNonReactObject = (obj: unknown): obj is AnyObject =>
-  _.isPlainObject(obj) && !(obj as AnyObject).$$typeof;
-
 /**
  * Recursive equivalent of {@link _.forOwn} function that traverses plain objects and arrays.
  */
 export const visitDeep = <TValue>(
   obj: AnyObject,
   predicate: (value: unknown) => value is TValue,
   valueCallback: (value: TValue, key: string, container: AnyObject) => void,
-  isPlainObject = isPlainNonReactObject,
+  isPlainObject: (obj: unknown) => obj is AnyObject = (o): o is AnyObject => _.isPlainObject(o),
```

Comment:
FYI - can't do
```ts
isPlainObject: (obj: unknown) => obj is AnyObject = _.isPlainObject,
```
since `_.isPlainObject` return type is just `boolean` (not a type guard).